modules/vaos/app/services/eps/provider_service.rb

```diff
@@ -101,23 +89,167 @@ def get_provider_slots(provider_id, opts = {})
     end
 
     ##
-    # Search for provider services using NPI
+    # Search for provider services using NPI, specialty, and address
     #
     # @param npi [String] NPI number to search for
+    # @param specialty [String] Specialty to match (case-insensitive)
+    # @param address [Hash] Address object with :street1, :city, :state, :zip keys
     #
     # @return OpenStruct response containing the provider service where an individual provider has
-    # matching NPI, or nil if not found
+    # matching NPI, specialty, and address, or nil if not found
     #
-    def search_provider_services(npi:)
-      query_params = { npi: }
+    def search_provider_services(npi:, specialty:, address:)
+      query_params = { npi:, isSelfSchedulable: true }
       response = perform(:get, "/#{config.base_path}/provider-services", query_params, request_headers)
 
-      # NPIs are unique, so we expect either an empty array or an array with exactly one matching provider
-      response.body[:provider_services]&.first&.then { |provider| OpenStruct.new(provider) }
+      return nil if response.body[:provider_services].blank?
+
+      # Filter providers by specialty and address
+      matching_provider = response.body[:provider_services].find do |provider|
+        specialty_matches?(provider, specialty) && address_matches?(provider, address)
+      end
+
+      matching_provider&.then { |provider| OpenStruct.new(provider) }
     end
 
     private
 
+    ##
+    # Check if provider's specialty matches the requested specialty (case-insensitive)
+    #
+    # @param provider [Hash] Provider data from EPS response
+    # @param specialty [String] Requested specialty to match against
+    # @return [Boolean] True if specialty matches, false otherwise
+    #
+    def specialty_matches?(provider, specialty)
+      return false if provider[:specialties].blank? || specialty.blank?
+
+      provider[:specialties].any? do |provider_specialty|
+        provider_specialty[:name].to_s.casecmp?(specialty.to_s)
+      end
+    end
+
+    ##
+    # Check if provider's address matches the requested address using a two-step approach
+    #
+    # @param provider [Hash] Provider data from EPS response
+    # @param address [Hash] Address object with :street1, :city, :state, :zip keys
+    # @return [Boolean] True if address matches, false otherwise
+    #
+    def address_matches?(provider, address)
```

Comment:
Note: Please chime in on this one. I tried to make this as flexible as possible, so we don't have to deal with format variations.

Reply:
Uses more broad qualifiers to exit early, reads logically, no notes from me 👍🏼